dali/imgcodec/decoders/nvjpeg2k/nvjpeg2k.cc

```diff
@@ -241,6 +242,10 @@ DecodeResult NvJpeg2000DecoderInstance::DecodeImplTask(int thread_idx,
   auto decode_out = out;
   if (is_processing_needed) {
     int64_t type_size = dali::TypeTable::GetTypeInfo(ctx.pixel_type).size();
+    size_t new_size = volume(ctx.shape) * type_size;
+    if (new_size > res.intermediate_buffer.capacity()) {
+      CUDA_CALL(cudaStreamSynchronize(ctx.cuda_stream));
```

Comment:
Why do we need this? Doesn't the stream-aware allocator make sure that it just works?

Reply:
`DeviceBuffer` isn't stream-aware.

Reply:
Understood.